exercises/practice/satellite/satellite.spec.js

```diff
@@ -0,0 +1,51 @@
+import { treeFromTraversals } from './satellite';
+
+describe('Satellite', () => {
+  test('Empty tree', () => {
+    expect(treeFromTraversals([], [])).toEqual({});
+  });
+
+  test('Tree with one item', () => {
+    const expected = { value: 'a', left: {}, right: {} };
+    expect(treeFromTraversals(['a'], ['a'])).toEqual(expected);
+  });
+
+  test('Tree with many items', () => {
+    const preorder = ['a', 'i', 'x', 'f', 'r'];
+    const inorder = ['i', 'a', 'f', 'x', 'r'];
+    const expected = {
+      value: 'a',
+      left: { value: 'i', left: {}, right: {} },
+      right: {
+        value: 'x',
+        left: { value: 'f', left: {}, right: {} },
+        right: { value: 'r', left: {}, right: {} },
+      },
+    };
+    expect(treeFromTraversals(preorder, inorder)).toEqual(expected);
+  });
+
+  test('Reject traversals of different length', () => {
+    const preorder = ['a', 'b'];
+    const inorder = ['b', 'a', 'r'];
+    expect(() => {
+      treeFromTraversals(preorder, inorder);
+    }).toThrowError('traversals must have the same length');
```

Comment:
* From the documentation, it looks the code you wrote means the error message somewhere contains the provided string. We want to check the error/message matches exactly. See last box in this section for how to set this up: https://jestjs.io/docs/expect#tothrowerror
* In most of the other tests we use the `toThrow` version instead of `toThrowError` (they do the same thing), maybe it makes sense to stick with that.

Reply:
Sorry, there was a misunderstanding here. With the link to the documentation, I just meant to go from
`toThrowError('traversals must have the same length');`
which only checks the message is contained somewhere in the error message to
`toThrow(new Error('traversals must have the same length'));`
which checks the error messages are actually the same

I didn't see there was a completely different example with custom error classes at the end of that example box. Sorry for not being more precise here earlier.

Reply:
No problem.
I should have asked for clarifications before implementing the code.
I'll fix the issue and submit the changes.